File: java-app/src/main/java/com/prosoft/java/repository/UserRepository.java
package com.prosoft.java.repository;

import com.prosoft.java.domain.User;
import org.springframework.stereotype.Repository;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

@Repository
public class UserRepository {

    private final List<User> users = new ArrayList<>();

    public UserRepository() {
        users.add(new User(1L, "User1", "user1@java.com"));
        users.add(new User(2L, "User2", "user2@java.com"));
        users.add(new User(3L, "User3", "user3@java.com"));
    }

    public List<User> findAll() {
        return users;
    }

    public Optional<User> findById(Long id) {
        return users.stream()
                .filter(user -> user.getId().equals(id))
                .findFirst();
    }

    public void save(User user) {
        users.add(user);
    }

    public void deleteById(Long id) {
        users.removeIf(user -> user.getId().equals(id));
    }

}

File: java-app/src/main/kotlin/com/prosoft/kotlin/repository/UserRepository.kt
package com.prosoft.kotlin.repository

import com.prosoft.kotlin.domain.User
import org.springframework.stereotype.Repository

@Repository("kotlinUserRepository")
class UserRepository {

    private val users = mutableListOf(
        User(1L, "User1", "user1@kotlin.com"),
        User(2L, "User2", "user2@kotlin.com"),
        User(3L, "User3", "user3@kotlin.com")
    )

    fun findAll(): List<User> = users

    fun findById(id: Long): User? {
        return users.find { it.id == id }
    }

    fun save(user: User) {
        users.add(user)
    }

    fun deleteById(id: Long) {
        users.removeIf { it.id == id }
    }
}
